tests/core/test_memory.py

```diff
@@ -101,6 +101,29 @@ def forward(self, x):
         return self.layer2(self.layer1(x))
 
 
+class LazyModel(LightningModule):
+    """ A model which contains lazy layers with unintialized parameters. """
+
+    def __init__(self):
+        super().__init__()
+        self.layer1 = nn.LazyLinear(5)
+        self.layer2 = nn.LazyLinear(2)
+
+    def forward(self, inp):
+        return self.layer2(self.layer1(inp))
+
+
+@RunIf(min_torch="1.8")
```

Comment:
Adding the min torch decorator still runs the tests for earlier versions of torch.

Reply:
The other tests are failing for earlier version of torch because the `memory.py` file now has an extra check for the `uninitialized parameter` which does not exist in older versions. So even if those additional tests are not run all other tests which check the summary would break.

Reply:
Additionally the torch version 1.9 test fails because the assertions don't match. Like we saw earlier the bias was always assigned as a normal parameter so was counted in 1.8, I guess they fixed it in 1.9 because the returned value of number of parameters is 0.

Reply:
addressed!